docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,416 @@
+---
+title: "Migrate from Newtonsoft.Json to System.Text.Json - .NET"
+author: tdykstra
+ms.author: tdykstra
+ms.date: "12/20/2019"
+helpviewer_keywords: 
+  - "JSON serialization"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# How to migrate from Newtonsoft.Json to System.Text.Json
+
+This article shows how to migrate from [Newtonsoft.Json](https://www.newtonsoft.com/json) to [System.Text.Json](system-text-json-overview.md).
+
+ `System.Text.Json` focuses primarily on performance, security, and standards compliance. It lacks built-in features to handle some common scenarios, and some default behaviors differ from `Newtonsoft.Json`. For some scenarios, `System.Text.Json` has no built-in functionality, but there are recommended workarounds. For other scenarios, workarounds are impractical. If your application depends on a missing feature that has no workaround, consider delaying your migration to `System.Text.Json`. 
+
+<!-- For information about which features might be added in future releases, see the [Roadmap](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Text.Json/roadmap/README.md). [Restore this when the roadmap is updated.]-->
+
+Most of this article is about how to use the <xref:System.Text.Json.JsonSerializer> API, but it also includes guidance on how to use the <xref:System.Text.Json.JsonDocument> Document Object Model (DOM) and the <xref:System.Text.Json.Utf8JsonReader> and <xref:System.Text.Json.Utf8JsonWriter> API. The article is organized into sections in the following order:
+
+* Differences in default `System.Text.Json.JsonSerializer` behavior compared to `Newtonsoft.Json`.
```

Comment:
I kept all the section headings at h2 instead of h2s for each grouping with h3s under them, because h2 headings automatically show up in the right-side navigation menu, making it easy to scan for a particular scenario you're interested in.  But that is probably more confusing than helpful, so I'll go back to h3/h2.

Reply:
Maybe add the headings with links as sort of a Table of Contents in the beginning (i.e. expand this bullet list with all the sub-sections). That can sort of substitute in for the R.H.S navigation menu but still keep the heading structure clear.

Reply:
>Is there no way to configure what shows up in the navigation menu?

😢 No, but there's an outstanding feature request.

Manually making a sub-TOC for each h2 section is feasible, but it has to be manually maintained and over time can get out of sync if updates are missed.